### ISSUE TITLE
commands not working when running in a Jupyter Notebook in VScode

### ISSUE THREAD

**Issue by samuelhaysom** [man-group/dtale#548]
I am trying to use dtale within a Jupiter notebook running inside VScode and am finding some commands within dtale don't do anything.

The commands I have found that don't work are, there are probably others that I haven't found yet
Code Snippet from the main Dtale menu
export as .csv from the main Dtale menu
Load data from main Dtale menu

Is this a bug, or is it due to how Jupiter notebooks are implemented in VScode as opposed to using them through a web browser?

**Comment by aschonfeld** [comment]
@samuelhaysom unfortunately this is a known bug with how vscode handles notebooks. See this [issue](https://github.com/man-group/dtale/issues/258)

That being said you can still run the following:
```
import dtale
import pandas as pd

df = pd.DataFrame([1,2,3,4,5])
d = dtale.show(df)
print(d._main_url)
```

Then take the output of `d._main_url` and paste it into a browser.  Please let me know if you have any ideas for a workaround (maybe something like an option to convert popups to redirects when in vscode). 
 Hope this helps and if you haven't already please toss your ⭐  on the repo.  Thanks 🙏

**Comment by samuelhaysom** [comment]
I see, makes sense. I would have thought you could add something so if dtale knows its being used in VScode it changes the way popups are generated to make vscode generate a new file with the output but I'm definitely not skilled enough to implement that.

It's a pity because being able to use notebooks from within VScode is such a nifty feature but I understand you can't support all use cases. I'll just continue to use notebooks in browser instead.

**Comment by aschonfeld** [comment]
I’ll see if i can dig around and see if theres a way to tell if you’re running a notebook from vscode or not. If there is then i can certainly implement that feature.

**Comment by aschonfeld** [comment]
Ok so it looks like I can execute `os.environ.get('VSCODE_PID') is not None` and if that statement is true then we're executing in VSCODE

**Comment by aschonfeld** [comment]
Good news, I've significant progress on this change and I should have a new version released in the next few days

**Comment by samuelhaysom** [comment]
Great! Thanks so much for having a look at this. I look forward to testing the new release :)

**Comment by aschonfeld** [comment]
@samuelhaysom just released v1.55.0 with this functionality, let me know if you still have issues
